docs(bpmn-copilot): add example prompts (#5046)
camunda/camunda-docs: docs/components/early-access/alpha/bpmn-copilot/bpmn-copilot.md

:::

### Example Prompts

- "Create an absence request diagram"
- "Design a process that uses data objects to drive process logic, showing how data influences process flow and decisions"
- "Model a complex service orchestration process involving multiple service tasks, conditional routing, and parallel execution of tasks."
- "Create a service order management process for a telco company. The process starts with a fraud check. First, calculate price dimensions via a script task, then check if the price dimensions are normal or high with an exclusive gateway. If high, escalate to a fraud investigation subprocess."
- Paste existing text documentation of a process (e.g., a Confluence page)
- Paste a process hard-coded in any language (e.g., BPEL, Java, COBOL, Python)

> - Paste a description of an diagram. This can be generated by any modern LLM like ChatGPT from a screenshot or picture

Author: ```suggestion
- Paste a description of an diagram. Any LLM (like ChatGPT) can generate this description from a screenshot or an image of the diagram.
```

- "Model a complex service orchestration process involving multiple service tasks, conditional routing, and parallel execution of tasks."
- "Create a service order management process for a telco company. The process starts with a fraud check. First, calculate price dimensions via a script task, then check if the price dimensions are normal or high with an exclusive gateway. If high, escalate to a fraud investigation subprocess."
- Paste existing text documentation of a process or of process requirements (e.g., a Confluence page)
- Paste a process hard-coded in any language (e.g., BPEL, Java, COBOL, Python)
- Paste a description of a diagram. Any LLM (like ChatGPT) can generate this description from a screenshot or an image of the diagram.

## Follow-Up Prompts:

The BPMN Copilot can also translate business-focused intent into actionable changes to the diagram it creates. For example:

- "Modify this process to maximize its changes of adhering to a 1 day SLA"
- "Reduce the operating cost of this process"
- "Show me ways to integrate AI into the process"
- "Consider unhappy paths as well"
- "Add error handling"
- "Improve the user experience"

> Note that the BPMN Copilot may not replicate all parts of the diagram exactly between modifications.

Reviewer: What do you mean by this?

Author: If you send a diagram A > B > C > D and ask to modify part B, the returned diagram could return unintended changes to part C.

Another phrasing is “the model might not return a 1 to 1 version of the diagram”

Does that make more sense? Happy to clarify if not😊

> Note that the BPMN Copilot may not replicate all parts of the diagram exactly between modifications.

Author: In general looks good! What about this word change?

```suggestion
:::note
Requesting specific modifications to one or several sections of the BPMN diagram may impact unrelated sections.
:::
```

> Note that the BPMN Copilot may not replicate all parts of the diagram exactly between modifications.

Reviewer: Love it!

- "Design a process that uses data objects to drive process logic, showing how data influences process flow and decisions"
- "Model a complex service orchestration process involving multiple service tasks, conditional routing, and parallel execution of tasks."
- "Create a service order management process for a telco company. The process starts with a fraud check. First, calculate price dimensions via a script task, then check if the price dimensions are normal or high with an exclusive gateway. If high, escalate to a fraud investigation subprocess."

> - Paste existing text documentation of a process (e.g., a Confluence page)

Reviewer 2: ```suggestion
- Paste existing text documentation of a process (e.g., a Confluence page)
- Paste a process hard-coded in any language (e.g. BPEL, Java, COBOL, Python)
- Paste a description of an diagram. This can be generated by any modern LLM like ChatGPT from a screenshot or picture
```

> 

Reviewer 2: ```suggestion
## Follow-Up Prompts:
The BPMN Copilot can also translate business-focused intent into actionable changes to the diagram. For example:
- "Modify this process to maximize its changes of adhering to a 1 day SLA"
- "Reduce the operating cost of this process"
- "Show me ways to integrate AI into the process"
- "Consider unhappy paths as well"
- "Add error handling"
- "Improve the user experience"

```

Author: While the BPMN Copilot can do some of these as expected with basic diagrams, I would hesitate to include them as examples yet.

I think it's important we set expectations appropriately for the BPMN Copilot at this early stage.

WDYT? Have you/customers had success with these prompts?

Reviewer 2: I had a demo with a customer where I said  "integrate AI into the process" and it blew them away
We should set expectations that it works well ONLY with the diagrams it creates, but otherwise it might time out

Author: @DanielFG-camunda @mathieu-stennier would you feel comfortable adding these "Follow up Prompts"?
